awsconfigrules/index_iam_user_policy_checker.js

```diff
@@ -0,0 +1,95 @@
+exports.handler = function(event, context ) {
+
+
+    const EVALUATION_TYPES = {
+        COMPLAINT: 'COMPLIANT',
+        NON_COMPLIANT: 'NON_COMPLIANT',
+    }; 
+
+    var getEvaluation = function getEvaluationParam(userId,value,timeStamp) {
+        var eval = [];
+        complianceType= EVALUATION_TYPES.COMPLAINT
+        if(value > 0){
+            complianceType= EVALUATION_TYPES.NON_COMPLIANT
+        }
+        eval.push({
+            ComplianceResourceType: "AWS::IAM::User" ,
+            ComplianceResourceId: userId,
+            ComplianceType: complianceType,
+            OrderingTimestamp: timeStamp,
+        });
+        return eval;
+    }
+
+    var aws_sts = new (require('../lib/aws/sts2.js'))();
+    var aws_ec2 = new (require('../lib/aws/ec2.js'))();
+    var aws_config = new (require('../lib/aws/awsconfig.js'))();
+    if (event.ruleParameters){
+        var ruleParameters = JSON.parse(event.ruleParameters);
+        event.federateRoleName = ruleParameters.federateRoleName;
+        event.federateAccount = ruleParameters.federateAccount;
+        event.account = ruleParameters.account;
+        event.roleName = ruleParameters.roleName;
+        event.roleExternalId = ruleParameters.roleExternalId;
+    }
+
+    if (!event.federateRoleName)  event.federateRoleName = "federate";
+
+    var roles = [];
+
+    if (event.federateAccount) {
+        roles.push({roleArn:'arn:aws:iam::' + event.federateAccount + ':role/' + event.federateRoleName});
+        var admin_role = {roleArn:'arn:aws:iam::' + event.account + ':role/' + event.roleName};
+        if (event.roleExternalId) {
+            admin_role.externalId = event.roleExternalId;
+        }
+        roles.push(admin_role);
+    }
+    console.log(roles);
+
+    var sessionName = event.sessionName;
+    if (sessionName == null || sessionName == "") {
+        sessionName = "session";
+    }
+
+    var invokingEvent = event.invokingEvent;
+    if (!ruleParameters) ruleParameters = {"region": event.region};
+
+    if (invokingEvent) invokingEvent = JSON.parse(invokingEvent);
+    else invokingEvent = {"configurationItem": {"resourceType": "IAM USER", "configurationItemCaptureTime": new Date()}};
+
+    if(event.resultToken) var resultToken = event.resultToken;
+    else var resultToken = "110ec58a-a0f2-4ac4-8393-c866d813b8d1";
+
+    var input = {
```

Comment:
you may break down the input params for each aws functions because now you're using the promise.